Assets/VRTK/Scripts/VRTK_PlayerJump.cs

```diff
@@ -0,0 +1,366 @@
+        {
+            if (Mathf.Abs(dv) < baselineUpdateSensitivity)
+            {
+                jumping = false;
+                crouchStartTime = 0;
+                //DebugHelper("Ending jump - isFalling = false");
+                // Trick to reset playerPresence so player can jump again
+                playerPresence.StartPhysicsFall(Vector3.zero);
+            }
+        }
+
+        private void UpdateBaseline(float dv)
+        {
+            if (Mathf.Abs(dv) < baselineUpdateSensitivity)
+            {
+                baselineHeight = HMDTransform.position.y;
+                playerPresence.StartPhysicsFall(Vector3.zero);
+            }
+        }
+
+        private float UpdateHeightHistoryAndReturnVelocity(float headHeight)
+        {
+            // TODO - calculation of difference will be off around the y origin
+            var deltaHeight = lastHeight - headHeight;
+            lastHeight = headHeight;
+
+            // update the history and get a sum
+            float heightSum = 0;
+            for (int i = 0; i < historyLen - 1; i++)
+            {
+                heightSum += historicVelocity[i];
+                historicVelocity[i] = historicVelocity[i + 1];
+            }
+            heightSum += historicVelocity[historyLen - 1];
+            historicVelocity[historyLen - 1] = deltaHeight;
+
+            // Velocity, scaled for convenience, signreversed for sanity or lazyness or something
+            return (heightSum / historyLen) * 100 * -1;
+        }
+
+        private void DoJump(float jumpVelocity)
+        {
+            Vector3 forward = GetPlayerForward() * jumpVelocity;
+
+            Vector3 calcVelocity;
+            calcVelocity.x = forwardFactor * forward.x;
+            calcVelocity.y = upFactor * forward.y;
+            calcVelocity.z = forwardFactor * forward.z;
+
+            //DebugHelper("JUMP EVENT: " + calcVelocity.ToString());
+
+            playerPresence.StartPhysicsFall(Vector3.zero);
+
+            playerPresence.GetComponent<Rigidbody>().velocity = calcVelocity;
+        }
+
+        private Vector3 GetPlayerForward()
+        {
+            // necessaries for triangulation
+
+            //var headPos = VRTK_DeviceFinder.HeadsetTransform().position;
+            //var rhand = VRTK_DeviceFinder.GetControllerRightHand().transform.position;
+            //var lhand = VRTK_DeviceFinder.GetControllerLeftHand().transform.position;
+
+            // these two numbers shall always add to 1.
+            // float xAmount = 0.5f;
+            // float zAmount = 0.5f;
+
+            // OR 
+
+            // a simpler method :) - assumes an X/Z plane
+            // guess I should normalize this.?.
+            Vector3 gaze = HMDTransform.forward;
+            return new Vector3(gaze.x, 1, gaze.z);
+        }
+
+        private void OnApplicationQuit()
+        {
+            QuitLogging();
+        }
+
+        #region ### Private Logging and Debugging Methods ###
```

Comment:
I think it's probably best to remove custom logging, it's only going to confuse someone down the line :)